### PR TITLE
Feature/k 03 personen

### PR DESCRIPTION
**Pull request by felixcicatt**
This PR takes care of the 'Personen' part of [this issue](https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/issues/158).

The flow:
1. Choose data to search in
2. Choose a string to search for
3. Depending on the data to search in, either Klantregister or BRP is searched
4. Depending on the source that's been searched, for each record, the other source is searched based on the BSN, if present. The result is used to enrich the record.

Miscellaneous:
- Added a function to format a date in ISO format, because the existing way caused issues because of time zones.

### CODE REVIEW

File: src/features/klant/service-data-enricher.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { ServiceResult, type NotUndefined, type ServiceData } from "@/services";
```
Comment (lencodes) [inline src/features/klant/service-data-enricher.ts:1]: I find this service quite hard to follow and understand. Could we simplify this?

Maybe we could call about it.

File: src/features/klant/_dev_fix_klanten.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// ONLY use this during development, to fix existing test data without a subjectType
+import { fetchLoggedIn, throwIfNotOk, parseJson } from "@/services";
+import { KlantType } from "./types";
+
+const klantRootUrl = `${window.gatewayBaseUri}/api/klanten`;
+
+function getAllKlantenUrl() {
+  const url = new URL(klantRootUrl);
+  url.searchParams.set("limit", "999");
```
[the remark this answers is not in the document]
Comment (felixcicatt) [reply src/features/klant/_dev_fix_klanten.ts:9]: It's more like a wildcard. But come to think of it, it might be nicer to turn this into a while loop of sorts, so we don't have too much requests at the same time. Thanks

File: src/features/klant/KlantenOverzichtRow.vue [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    persoon.voorvoegselAchternaam,
+    persoon.achternaam,
+  ]
+    .filter(Boolean)
+    .join(" ");
+  return (
+    persoon && {
+      ...persoon,
+      naam,
+      adres: [persoon.postcode, persoon.huisnummer].filter(Boolean).join(" "),
```
[the remark this answers is not in the document]
Comment (felixcicatt) [reply src/features/klant/KlantenOverzichtRow.vue]: Agreed. Fixed

File: src/features/klant/KlantenOverzichtRow.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      : [undefined, props.record],
+  (k) => k.bsn,
+  (p) => p.bsn,
+  useKlantByBsn,
+  usePersoonByBsn
+);
+
+const getKlantUrl = (klant: Klant) => `/klanten/${klant.id}`;
+
+function mapKlant(klant?: Klant | null) {
```
[the remark this answers is not in the document]
Comment (felixcicatt) [reply src/features/klant/KlantenOverzichtRow.vue]: I did some refactoring to seperate the mapping logic. However, I didn't manage to pull it into klant-zoeker, because the enrichment can't be done inside of a map function (as far as my experiments showed). Does this look better?

File: src/features/klant/KlantenOverzichtRow.vue [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        <dutch-date
+          v-if="result.persoon.data?.geboortedatum"
+          :date="result.persoon.data.geboortedatum"
+        />
+      </td>
+      <td>{{ result.persoon.data?.adres }}</td>
+    </template>
+    <td colspan="2" v-else>
+      <simple-spinner class="spinner" v-if="result.persoon.loading" />
+    </td>
```
[the remark this answers is not in the document]
Comment (felixcicatt) [reply src/features/klant/KlantenOverzichtRow.vue]: Agreed. The UI still jumps, but this is a bit nicer. This opens the way for a bit nicer loading animations, skeleton or similar.

File: src/features/klant/KlantenOverzichtRow.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (lencodes) [reply src/features/klant/KlantenOverzichtRow.vue]: Much better!

**Review by lencodes — APPROVED** [review]
Very nice. Approved.
